test/e2e/app-dir/use-cache/app/with-server-action/layout.tsx

```diff
@@ -0,0 +1,14 @@
+import { connection } from 'next/server'
+
+export default async function DynamicLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  // TODO: This is a workaround for Turbopack. Figure out why this fails during
+  // prerendering with:
+  // TypeError: Cannot read properties of undefined (reading 'Form')
```

Comment:
The cause for this error is a race condition for setting the client manifest into the singleton, similar to what we fixed for the server manifest in #71669. Needs to be fixed in a follow-up.